src/main/java/models/CompositeKey.java

```diff
@@ -107,7 +108,12 @@ public void write(DataOutput dataOutput) throws IOException {
         if (this.getKeyType() == KeyType.DOCUMENT) {
             dataOutput.writeLong(this.getDocumentId().get());
         } else {
-            dataOutput.writeUTF(this.getTerm());
+            try {
+                dataOutput.writeUTF(this.getTerm());
+            }catch (UTFDataFormatException e){
+                // Truncate the term to 65535
+                dataOutput.writeUTF(new String(this.getTerm().getBytes(),0, 65535 - 2));
```

Comment:
What's the reasoning behind the -2? One for zero indexing, and one for safety?
```suggestion
                dataOutput.writeUTF(this.getTerm().substring(0, s.length() - 2));
```
(this suggestion avoids a new string allocation)